src/App.tsx

```diff
@@ -5,14 +5,21 @@ import AddTodoItemInput from "./AddTodoItemInput/AddTodoItemInput";
 import { getItemsFromCache } from "./utils";
 import { TodoItem } from "./types";
 import { Container } from "@mui/system";
-import { CssBaseline } from "@mui/material";
+import { CssBaseline, IconButton } from "@mui/material";
+import { useTheme, ThemeProvider, createTheme } from "@mui/material/styles";
+import useMediaQuery from "@mui/material/useMediaQuery";
 
 function App() {
   const [value, setValue] = useState("");
   const [todoItems, setTodoItems] = useState<TodoItem[]>(
     getItemsFromCache() || []
   );
 
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const [mode, setMode] = useState<"light" | "dark">(
```

Comment:
You don't need this intermediate state. You can save it to a simple variable like so: `const mode = prefersDarkMode  ? 'dark' : 'light`. And then pass it to ThemeProvider